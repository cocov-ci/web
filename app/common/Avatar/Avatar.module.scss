@use "sass:color";
@import "styles/variables";

.base {
  background: $colorAvatarBackground;
  box-shadow: 0 0 2px 0 color.adjust($colorBlack, $alpha: -0.5);
  border-radius: 50%;
}
