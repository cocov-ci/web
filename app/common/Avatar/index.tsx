import classNames from 'classnames'
import React from 'react'

import styles from './Avatar.module.scss'

type AvatarProps = {
  className?: string
  avatarURL: string
  size: string
}

const Avatar = ({ avatarURL, className, size }: AvatarProps) => {
  return (
    // eslint-disable-next-line @next/next/no-img-element
    <img
      alt=""
      aria-hidden={true}
      className={classNames(styles.base, className)}
      src={avatarURL}
      style={{
        height: size,
        width: size,
      }}
    />
  )
}

export default Avatar
