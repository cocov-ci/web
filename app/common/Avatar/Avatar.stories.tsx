import type { Meta, StoryObj } from '@storybook/react'

import Avatar from './index'

const meta: Meta<typeof Avatar> = {
  title: 'Common/Avatar',
  component: Avatar,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof Avatar>

export const Default: Story = {
  args: {
    avatarURL: 'https://avatars.githubusercontent.com/u/118852412?s=200&v=4',
    size: '64px',
  },
}
