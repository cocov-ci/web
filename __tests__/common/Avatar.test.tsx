import { render } from '@testing-library/react'

import Avatar from 'app/common/Avatar'
import '@testing-library/jest-dom'

describe('common/Avatar', () => {
  const { container } = render(
    <Avatar
      avatarURL="https://avatars.githubusercontent.com/u/118852412?s=200&v=4"
      size="30px"
    />,
  )

  it('renders Avatar', () => {
    expect(container.querySelector('img')?.getAttribute('src')).toBe(
      'https://avatars.githubusercontent.com/u/118852412?s=200&v=4',
    )
  })

  it('renders Avatar snapshots', () => {
    expect(container).toMatchSnapshot()
  })
})
