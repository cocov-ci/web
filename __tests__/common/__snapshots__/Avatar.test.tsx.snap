// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Avatar renders Avatar snapshots 1`] = `<div />`;
